### PR TITLE
[UT/Codecov] : Updates to UT for utils and code coverage path

### PR DESCRIPTION
**Pull request by sachin-apa**
# Description
Fix Code Coverage app not showing the file coverage on browsing.

# GitHub Issues
List the GitHub issues impacted by this PR:
UT added to utils redfish, ome, and idrac_redfish.
| GitHub Issue # |
| -------------- |
| |

# ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME

 redfish, ome, and idrac_redfish.
##### OUTPUT

```paste below

```
##### ADDITIONAL INFORMATION

```paste below

```
# Checklist:

- [x] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [x] I have verified that new and existing unit tests pass locally with my changes
- [x] I have not allowed coverage numbers to degenerate
- [x] I have maintained at least 90% code coverage
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have maintained backward compatibility

### CODE REVIEW

**Review by ABHISHEK-SINHA10 — DISMISSED** [review]

**Review by ShivamSh3 — DISMISSED** [review]

**Review by felixs88 — DISMISSED** [review]

**Review by Kritika-Bhateja-03 — DISMISSED** [review]

**Review by jagadeeshnv — DISMISSED** [review]

File: tests/unit/plugins/module_utils/test_idrac_redfish.py [hunk 1]
```diff
@@ -0,0 +1,383 @@
...
+        mock_response.getcode.return_value = 200
+        mock_response.headers = mock_response.getheaders.return_value = {
+            'X-Auth-Token': 'token_id'}
+        mock_response.read.return_value = json.dumps({"value": "data"})
+        return mock_response
+
+    def test_invoke_request_with_session(self, mock_response, mocker):
+        mocker.patch(MODULE_UTIL_PATH + OPEN_URL,
+                     return_value=mock_response)
+        module_params = {'idrac_ip': '192.168.0.1', 'idrac_user': 'username',
```
Comment (Jennifer-John) [inline tests/unit/plugins/module_utils/test_idrac_redfish.py]: can we declare module_params as part of fixture? most methods seem to have same defined?

File: tests/unit/plugins/module_utils/test_ome.py [hunk 2]
```diff
@@ -215,70 +225,250 @@ def test_get_device_id_from_service_tag_ome_case02(self, mocker, mock_response):
...
+            ret = obj.test_network_connection(
+                "share_address", "share_path", "share_type", "share_user", "share_password", "share_domain")
+        assert ret.json_data == mock_response.json_data
+
+        with RestOME(module_params, True) as obj:
+            ret = obj.test_network_connection(
+                "share_address", "share_path", "share_type")
+        assert ret.json_data == mock_response.json_data
+
+    # @pytest.mark.parametrize("ret_val", [
```
Comment (Jennifer-John) [inline tests/unit/plugins/module_utils/test_ome.py]: pls, remove commented code

Comment (sachin-apa) [reply tests/unit/plugins/module_utils/test_ome.py]: removed

File: tests/unit/plugins/module_utils/test_idrac_redfish.py (hunk 1 above)
Comment (sachin-apa) [reply tests/unit/plugins/module_utils/test_idrac_redfish.py]: updated to refactor the module params and conn obj to fixtures

File: tests/unit/plugins/module_utils/test_idrac_redfish.py [hunk 3]
```diff
@@ -0,0 +1,345 @@
+# -*- coding: utf-8 -*-
+
+#
+# Dell OpenManage Ansible Modules
+# Version 8.3.0
+# Copyright (C) 2013 Dell Inc.
```
Comment (Jennifer-John) [inline tests/unit/plugins/module_utils/test_idrac_redfish.py]: 2023? applicable to all instances.

File: tests/unit/plugins/module_utils/test_ome.py [hunk 4]
```diff
@@ -17,268 +17,404 @@
...
-    def test_get_all_items_with_pagination_error_case(self, mock_response, mocker):
-        mocker.patch(MODULE_UTIL_PATH + 'ome.open_url',
+        mock_response_page2 = MagicMock()
+        mock_response_page2.success = True
+        mock_response_page2.status_code = 200
+        mock_response_page2.json_data = {
+            ODATA_COUNT: 100, "value": list(range(50, 100))}
+
+        def mock_invoke_request(*args, **kwargs):
+            print(args)
```
Comment (Jennifer-John) [inline tests/unit/plugins/module_utils/test_ome.py]: we can remove the print stmt?

**Review by Jennifer-John — APPROVED** [review]

File: tests/unit/plugins/module_utils/test_idrac_redfish.py (hunk 3 above)
Comment (sachin-apa) [reply tests/unit/plugins/module_utils/test_idrac_redfish.py]: updated

File: tests/unit/plugins/module_utils/test_ome.py (hunk 4 above)
Comment (sachin-apa) [reply tests/unit/plugins/module_utils/test_ome.py]: removed

**Review by ABHISHEK-SINHA10 — APPROVED** [review]

**Review by Kritika-Bhateja-03 — APPROVED** [review]

**Pull request merged**